### PR TITLE
feat(dive-log): collapsible trip groups in the dive list (#1193)

### LINKED ISSUE #1193: Display trip groupings in-line in the dive list
**Is your feature request related to a problem? Please describe.**
More easily navigate through the dive list in an intuitive manner

**Describe the solution you'd like**
Show collapsible view of trips in-line in the dive list view. See mock-up below. Clicking on the trip header toggle expanding/collapsing the dives that are included on the trip.

Trip expanded:
Image

Trip collapsed:
Image

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

Closes #1193. Consecutive dives that belong to the same trip now fold under a
collapsible trip header in the Dives list, so a 14-dive liveaboard can be
tidied away to reach the dives on either side.

The header pins to the top of its own group while you scroll through it and
releases at the next group, and it reads "6 of 14 dives" when only part of a
trip has paged in, so a trip cut by the 50-dive page boundary does not look
like a smaller trip.

**Grouping never makes a dive card narrower.** The mock-ups on the issue indent
trip dives behind a rail; that was rejected. Instead the group is signalled by
the header and by a faint band painted *behind* it with `DecoratedSliver`, so
the grouped region is wider than the cards rather than narrower. Two regression
tests assert a grouped and a loose card have identical width and identical left
edge.

Off by default: it is a sticky per-diver setting under the list's overflow menu,
alongside Expand all trips and Collapse all trips.

## Changes

- `DiveSummary` gains four trip scalars, fed by a `LEFT JOIN trips` added to
  both summary SELECTs. No `Trip` object is hydrated: the entity exists to keep
  list rows slim.
- `getTripDiveCounts()` plus `tripDiveCountsProvider`: one grouped COUNT for the
  whole list, not one per header, so the "of 14" is affordable.
- Schema **v204**, `diver_settings.group_trips_in_dive_list`. Collapsed trip ids
  live in SharedPreferences instead, keyed per diver, since which trips you
  folded away is per-device browsing state and does not belong in sync.
- `buildDiveListSections()`, a pure function splitting the loaded page into
  loose runs and trip runs. Sections are runs, not sets: a trip interrupted by a
  loose dive yields two sections, both labelled with the trip's real total,
  because gathering a trip out of date order would break the sort.
- The list becomes a `CustomScrollView`; each trip is a `DecoratedSliver`
  wrapping a `SliverMainAxisGroup` of a pinned header and its rows. Rows still
  render through the shared `DiveListItem`, untouched.
- Grouping is gated on the toggle, a chronological sort and a card view mode.
  Under a depth sort the headers disappear and an inline notice says why, with a
  "Sort by date" action. The toggle never silently reverts. Table mode ignores
  grouping entirely.
- Selection mode puts a tri-state checkbox on the header covering the group's
  loaded dives; a shift-range skips dives folded inside a collapsed trip.
- 11 new l10n keys across all 11 locales.

### Note for reviewers

The pagination sentinel is deliberately a built widget whose `build` kicks the
next page, not a scroll-offset check (#1610). It matters more now: collapse
every trip and a freshly loaded page of 50 can be almost entirely invisible,
with no scroll event to notice it. There is a test asserting the list keeps
paging in that state and stops once the source is drained.

## Test Plan

- [x] `flutter test` passes (26,521 passed, 21 skipped, 0 failures)
- [x] `flutter analyze --fatal-infos` passes
- [x] `flutter gen-l10n` produces no diff, so the checked-in generated l10n is current
- [x] Manual testing on: **iPhone 17 Pro simulator, iOS 26**, light and dark.
      Verified by eye rather than inferred from tests:
      - grouped and loose dive cards share the same left edge and width, which
        is the constraint this PR exists to respect
      - the trip header pins to the top of its own group while its dives
        scroll underneath, and releases at the next group
      - collapse folds the trip and survives a full app kill and relaunch, as
        does the Group trips toggle
      - the header count follows the trip, updating from 3 to 15 dives when
        the trip grew
      - the Group trips menu entry matches the view-mode entries: leading
        icon, no checkbox, tint tracking its state

The band shading was tuned on device, not guessed at: 0.14 read as nothing,
0.38 was still not a band, and it took close to the full container colour to
group as one block. The reason a small bump did not help is structural, since
the near-white cards nearly fill the band and only the margins show any tint.

Schema note: this rung has been renumbered twice, 202 to 201 to **204**, as
#1677 (v200), the O2 cell linearity link (v201), condition intelligence (v202)
and equipment assemblies (v203) each landed while this branch was open. A rung
at or below the shipped version never runs its `onUpgrade` step, so if more
schema work merges before this does, it needs renumbering again.

An earlier revision of this branch also carried a fix for the bare `TextButton`
on the transmitter editor's app bar. #1701 and #1709 fixed it on main first, so
that commit has been dropped and the file now matches main exactly.

### CODE REVIEW

File: lib/features/dive_log/presentation/providers/trip_group_collapse_provider.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+///
+/// Device-local on purpose. Which trips a diver has tidied away is browsing
+/// state, not logbook data, so it stays out of the synced settings row rather
+/// than pushing an ever-growing id blob through sync (issue #1193).
+const String kCollapsedTripIdsPrefsKey = 'dive_list_collapsed_trip_ids';
+
+/// Trips folded shut in the dive list, persisted across restarts.
+class CollapsedTripsNotifier extends StateNotifier<Set<String>> {
+  CollapsedTripsNotifier(this._prefs)
+    : super(_prefs.getStringList(kCollapsedTripIdsPrefsKey)?.toSet() ?? {});
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/presentation/providers/trip_group_collapse_provider.dart]: Good catch, and the spec called for exactly this: "a Set<String> in SharedPreferences, keyed per diver". I shipped a single global key.

Fixed in a37e7e5. The key is now suffixed with the diver id via collapsedTripIdsPrefsKeyFor(), and collapsedTripIdsProvider watches currentDiverIdProvider so switching profiles rebuilds the notifier against that diver's own set rather than inheriting the previous one's folded trips. The unsuffixed key is kept for the no-active-diver case rather than inventing a second shape for it.

Two tests added: one diver's collapse does not fold the trip for another, and each diver keeps their own set across container rebuilds.

File: lib/features/dive_log/presentation/widgets/dive_list_content.dart [hunk 2]
```diff
@@ -1540,62 +1603,246 @@ class _DiveListContentState extends ConsumerState<DiveListContent> {
...
-                  }
-                  _loadNextPageIfStranded(paginatedState);
-                  return const Padding(
-                    padding: EdgeInsets.symmetric(vertical: 16),
-                    child: Center(child: CircularProgressIndicator()),
+              slivers: [
+                if (ref.watch(diveListGroupTripsProvider) && !groupingEnabled)
+                  SliverToBoxAdapter(
+                    child: _buildGroupingPausedNotice(context),
+                  ),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/presentation/widgets/dive_list_content.dart]: The premise is not quite right, but the underlying worry is, so I changed it anyway.

Table mode never reaches this code: build() returns _buildTableModeScaffold before _buildDiveList is called, so the notice cannot render there today. I confirmed that rather than assuming it, and an attempted widget test for it could not even be pumped, since table mode uses the real repository path and needs a database the harness does not provide.

But that correctness was resting on an early return about 800 lines away, which is too distant a guarantee. The notice now reads a dedicated diveListGroupingPausedBySortProvider that ignores the view mode entirely, so it is true only when the sort really is the blocker. Four provider tests pin it, including both table-mode cases.

File: test/features/dive_log/presentation/widgets/dive_list_grouping_menu_test.dart [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      grouping: grouping,
+      tripTotals: const {'t1': 1},
+      viewMode: viewMode,
+    );
+
+    await tester.pumpWidget(
+      testApp(
+        overrides: overrides,
+        child: const DiveListContent(showAppBar: true),
+      ),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply test/features/dive_log/presentation/widgets/dive_list_grouping_menu_test.dart:29]: Agreed, fixed in a37e7e5: testApp(locale: const Locale('en')).

Applied to all five grouping test files rather than only the three flagged here, since dive_list_trip_grouping_test.dart and its selection companion assert on site names and dive counts through the same harness.

File: test/features/dive_log/presentation/widgets/dive_list_grouping_paused_test.dart [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:submersion/core/constants/sort_options.dart';
+import 'package:submersion/core/models/sort_state.dart';
+import 'package:submersion/features/dive_log/presentation/providers/dive_providers.dart';
+import 'package:submersion/features/dive_log/presentation/widgets/dive_list_content.dart';
+import 'package:submersion/features/dive_log/presentation/widgets/trip_group_header.dart';
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply test/features/dive_log/presentation/widgets/dive_list_grouping_paused_test.dart:8]: Agreed, fixed in a37e7e5. Both call sites in this file now pass locale: const Locale('en'), which matters most here because the sort-field name in the notice ("Max Depth") is itself localized.

File: test/features/dive_log/presentation/widgets/trip_group_header_test.dart [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        overrides: overrides,
+        child: TripGroupHeader(
+          section: value,
+          onToggle: onToggle ?? () {},
+          onOpenTrip: onOpenTrip ?? () {},
+          isSelectionMode: isSelectionMode,
+          groupChecked: groupChecked,
+          onGroupCheckedChanged: onGroupCheckedChanged,
+        ),
+      ),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply test/features/dive_log/presentation/widgets/trip_group_header_test.dart:82]: Agreed, fixed in a37e7e5. This one was the most exposed of the three: it asserts on a month abbreviation from UnitFormatter.formatDateRange as well as the "6 of 14 dives" string, so an unpinned locale could fail it two different ways.

File: lib/features/dive_log/data/repositories/dive_repository_impl.dart [hunk 6]
```diff
@@ -2804,6 +2820,28 @@ class DiveRepository {
...
+  /// comparison has to ignore the view filter. Trips with no dives are absent
+  /// rather than zero, which is what the header wants anyway.
+  // stats-scope-exempt: a structural count for list chrome, not a statistic.
+  Future<Map<String, int>> getTripDiveCounts() async {
+    final rows = await _db
+        .customSelect(
+          'SELECT trip_id, COUNT(*) AS n FROM dives '
+          'WHERE trip_id IS NOT NULL GROUP BY trip_id',
+          readsFrom: {_db.dives},
+        )
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/data/repositories/dive_repository_impl.dart:2908]: Valid, fixed in e700acb. getTripDiveCounts() now takes an optional diverId and appends `diver_id = ?`, matching the convention every other list query in this repository already uses (`d.diver_id = ?`). tripDiveCountsProvider passes the active diver from currentDiverIdProvider.

Both halves of your point mattered: on a shared library the header was counting another diver's dives into the "of 14", and it was reading rows the list would never show.

Two tests added: a trip with dives from two divers counts only the named one, and a null diver id still counts everything, so the unscoped call keeps its old meaning.

File: lib/features/dive_log/presentation/widgets/trip_group_header.dart [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+  bool shouldRebuild(TripGroupHeaderDelegate oldDelegate) {
+    return oldDelegate.section.tripId != section.tripId ||
+        oldDelegate.section.tripName != section.tripName ||
+        oldDelegate.section.collapsed != section.collapsed ||
+        oldDelegate.section.loadedCount != section.loadedCount ||
+        oldDelegate.section.totalCount != section.totalCount ||
+        oldDelegate.extent != extent ||
+        oldDelegate.isSelectionMode != isSelectionMode ||
+        oldDelegate.groupChecked != groupChecked;
+  }
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/presentation/widgets/trip_group_header.dart:237]: Agreed, and fixed in e700acb by returning true unconditionally.

You are right about the mechanism: every callback passed to this delegate is a fresh closure built by the list, and onGroupCheckedChanged captures the group's dive ids by value. Comparing only the visible fields could leave a stale closure acting on an outdated group, which is a correctness bug traded for skipping the rebuild of one small fixed-extent row. Not a good trade.

File: lib/features/dive_log/presentation/providers/trip_group_collapse_provider.dart [hunk 8]
```diff
@@ -0,0 +1,69 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+import 'package:submersion/core/providers/provider.dart';
+import 'package:submersion/features/divers/presentation/providers/diver_providers.dart';
+import 'package:submersion/features/settings/presentation/providers/settings_providers.dart';
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/presentation/providers/trip_group_collapse_provider.dart:8]: This one I do not think is right, so I have left it as is.

settings_providers.dart is where sharedPreferencesProvider is declared (settings_providers.dart:1013), and this file reads it on line 66 to construct the notifier. Removing the import would not compile.

Worth noting CI runs `flutter analyze --fatal-infos` over the whole project, which fails the build on an unused_import at info level, and it is green on this branch.

**Comment by ericgriffin** [comment]
The suppressed finding from the latest review (no thread to reply on) is fixed in 1dc4ef4.

> SharedPreferences writes are fire-and-forget here, but `_prefs.setStringList(...)` returns a Future that can complete with an error; leaving it unhandled can surface as an unhandled async exception.

Correct, and mine. The write stays unawaited, because every caller is a tap handler and folding a trip should not wait on disk, but the rejection is now caught and logged rather than left on a bare future. The in-memory set is the source of truth for the session, so a failed write costs one forgotten fold, not data.

`QualityDetectorTogglesNotifier` was a good pointer, though it awaits from an already-async setter; these methods are void and called straight from tap handlers, so explicit error handling fitted better than making the whole API async.

Covered by a test that drives it with a SharedPreferences whose writes always reject; reverting the handler makes that test fail, so it is pinning the mechanism rather than the shape.

File: lib/features/dive_log/presentation/widgets/dive_list_content.dart [hunk 9]
```diff
@@ -1540,62 +1603,256 @@ class _DiveListContentState extends ConsumerState<DiveListContent> {
...
-                  _loadNextPageIfStranded(paginatedState);
-                  return const Padding(
-                    padding: EdgeInsets.symmetric(vertical: 16),
-                    child: Center(child: CircularProgressIndicator()),
+              slivers: [
+                if (ref.watch(diveListGroupingPausedBySortProvider))
+                  SliverToBoxAdapter(
+                    child: _buildGroupingPausedNotice(context),
+                  ),
+                for (final section in sections)
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/presentation/widgets/dive_list_content.dart:1844]: Valid, and a real regression from the sliver conversion. Fixed in 5fd76a7.

Both halves of your point were right. The estimate divided the whole scrollable height by the loaded dive count, so the new non-dive slivers (a pinned header per trip, the paused notice, the bottom spacer) were counted as if they were dive rows; and it multiplied by the flat index, so a dive folded inside a collapsed trip counted despite occupying no height.

It now subtracts the chrome before averaging, counts the target's position among visible dives, and adds the extent of each header above it.

Worth recording the measurement, because my first attempt at a test was useless: with three expanded trips the old and new formulas differ by about 30px, under half a row, so the test passed against both. The error only becomes material once trips are collapsed. On a 120-dive list with every trip folded, selecting a dive near the end: the old formula lands the row at y=26, jammed against the top edge of a 600pt viewport, where the new one lands it at y=254, close to the intended third down. The test asserts that distance and fails at 26, so it is pinning the behaviour rather than the shape.

File: lib/features/dive_log/presentation/widgets/dive_list_content.dart [hunk 10]
```diff
@@ -1540,62 +1666,260 @@ class _DiveListContentState extends ConsumerState<DiveListContent> {
...
+  /// Not folded into the active-filters bar: that bar only exists when a
+  /// filter is active, and a paused grouping is not a filter. The toggle stays
+  /// on and stays checked, so nothing the diver set reverts behind their back
+  /// (#1193).
+  Widget _buildGroupingPausedNotice(BuildContext context) {
+    final theme = Theme.of(context);
+    final sort = ref.watch(diveSortProvider);
+    return Padding(
+      padding: const EdgeInsets.fromLTRB(16, 8, 8, 0),
+      child: Row(
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/presentation/widgets/dive_list_content.dart:1923]: Correct, fixed in f66d05c. I measured the widget rather than reasoning about the padding: it renders at 56, not 48. The Row is 48 (the action button's tap target) under 8px of top padding, so every scroll-to-selected estimate was biased by 8px whenever the notice was showing.

I took the second of your two options rather than the first, and it is worth saying why. Constraining the notice to the constant makes any test comparing the two a tautology: I tried exactly that first, wrapped it in a SizedBox of kGroupingPausedNoticeHeight and asserted the rendered height equalled the constant, and the test passed even with the constant set back to the wrong 48. It would also hide a future content change by clipping it rather than failing.

So the notice sizes itself, the constant records what it renders at, and the test measures the real widget against it. Setting the constant back to 48 now fails with Expected <48.0>, Actual <56.0>.

**Comment by ericgriffin** [comment]
Both suppressed findings from the latest review are fixed in 84bda00. Neither had a thread to reply on, so recording them here.

**Stale trip and site names in the list.** Correct, and the sharper half of the point is that I had made two mechanisms disagree: I declared `trips` in `readsFrom`, saying the list reads it, while the list actually reloads off `watchDivesChanges()`, which watches only the `dives` table. A trip rename therefore sat on screen stale until some unrelated dive write happened to shake the list.

New `watchDiveListChanges()` watches exactly the three tables the summary SELECT reads: `dives`, `dive_sites`, `trips`. Deliberately not `watchDiveDetailChanges()`, which already includes `trips` but also profile series, tank pressures, equipment and safety findings; none of those reach the list, and each would reload the whole thing for nothing.

This also fixes a pre-existing bug the grouping work only exposed: site renames were equally stale, for the same reason.

The regression test states the bug in one assertion: a trip rename ticks the list stream and provably does not tick the dives-only stream.

**Header extent capped at 2.0.** Also correct, and self-defeating: the fixed extent exists to stop the second line clipping, and the cap reintroduced exactly that for anyone past 200%, which both iOS and Android offer. It now grows without an upper bound and only holds the lower one, so a 300% reader gets a taller header rather than a clipped one. Restoring the cap fails the new test with `Expected a value greater than <124.0>, Actual <124.0>`.

Worth flagging for anyone touching `DiveRepository`: adding one method broke test doubles in three different ways, and none of them surfaces the others. Hand-rolled `implements` doubles fail `flutter analyze`; generated `.mocks.dart` files need `build_runner`; and strict `when(...)` mocks throw MissingStubError at runtime, which showed up as an unrelated provider never loading.

File: lib/features/dive_log/data/repositories/dive_repository_impl.dart [hunk 11]
```diff
@@ -151,6 +151,30 @@ class DiveRepository {
...
+  /// nothing.
+  Stream<void> watchDiveListChanges() => _db
+      .tableUpdates(
+        TableUpdateQuery.allOf([
+          TableUpdateQuery.onTable(_db.dives),
+          TableUpdateQuery.onTable(_db.diveSites),
+          TableUpdateQuery.onTable(_db.trips),
+        ]),
+      )
+      .debounce(changeTickDebounce);
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/data/repositories/dive_repository_impl.dart]: Correct, fixed in b278c14. dive_safety_findings is now in the tick, so a synced or freshly swept review updates the row badge instead of waiting for an unrelated dive, site or trip write.

Checking against the query's own readsFrom showed my tick was short by three tables, not one: dive_profile_series and dive_profile_events are in there too, because the decompression filter joins them via decoSignalCondition. I have left those two out on purpose and said so in the doc rather than let the comment overclaim a second time. A profile import writes thousands of sample rows, and ticking on each would reload the whole list behind every one of them, to correct a filter most sessions never switch on. That gap is no worse than the dives-only tick this replaced.

Mutation-checked: dropping dive_safety_findings from the tick fails the new test. The test inserts through the generated Drift companion rather than literal SQL, after my first attempt failed on an updated_at column this table does not have.

**Comment by ericgriffin** [comment]
The suppressed finding about the scroll giving up on a collapsed trip is fixed in ca3b7f5. No thread to reply on, so recording it here.

The diagnosis was right, and the merge path is where it actually bites: `_scrollToSelectedItem` measures the layout of the build it runs after, and a merge passes its dive as an `overrideId`, which never reaches `widget.selectedId` and so never forced any trip open. The scroll measured a build where the dive was invisible, returned, and never tried again.

The scroll target now feeds the same force-expand slot as the open dive, and the scroll retries once on the far side of the rebuild. Bounded to one attempt per target: a dive still invisible after its trip is forced open will not become visible by trying again, and an unbounded retry would ping-pong between build and post-frame forever.

On coverage, so the seam is visible rather than implied. I could not drive this through a widget test, and the control experiment says why: registering a bare `addPostFrameCallback` from the test itself does not fire either, because no post-frame callback runs once the tree has settled and nothing schedules a frame. Rather than ship the judgement untested, it is extracted as a pure `shouldRetryScrollAfterExpanding` with six unit tests covering retry-when-folded, and skip when already visible, not loaded, loose, or already retried, plus that a retry spent on one dive does not block another. Removing the bound fails those tests. The frame plumbing around the decision is not covered.

**Comment by ericgriffin** [comment]
Both suppressed findings are fixed in a483774. No threads to reply on, so recording them here.

**_scrollTargetId never cleared.** Correct, and worse than described: it was a user-visible bug I introduced one commit earlier, where tapping a trip header to collapse it appeared to do nothing.

The obvious fix is also wrong, which is what makes it worth writing down. Clearing the field after a successful scroll re-collapses the trip on the next build and hides the dive the scroll just travelled to, because the trip was only open by virtue of the override. The override was the wrong mechanism rather than one missing a cleanup step, so it is gone. The retry now expands the trip for real through a new `CollapsedTripsNotifier.expand()`. Nothing overrides the rendered state, so the header control keeps working and the diver can fold it back: a visible, undoable action instead of a hidden one.

**Blank trip identity from optimistic summaries.** Also correct. `DiveSummary.fromDive` takes `tripId` from `dive.tripId` but the name and dates only from `dive.trip`, so `updateDive` and `addDive` can produce a summary that knows its trip id and nothing else, and a run headed by one blanked the whole header until the next database read. Section identity now comes from the first entry that carries it rather than simply the first entry. A run of only optimistic summaries still groups on the trip id, with an empty name, rather than throwing or splitting.

Both mutation-checked: reverting either fix fails its test.

File: lib/features/dive_log/data/repositories/dive_repository_impl.dart [hunk 12]
```diff
@@ -151,6 +151,40 @@ class DiveRepository {
...
+  /// filter most sessions never switch on. This is no worse than the
+  /// dives-only tick it replaces.
+  Stream<void> watchDiveListChanges() => _db
+      .tableUpdates(
+        TableUpdateQuery.allOf([
+          TableUpdateQuery.onTable(_db.dives),
+          TableUpdateQuery.onTable(_db.diveSites),
+          TableUpdateQuery.onTable(_db.trips),
+          TableUpdateQuery.onTable(_db.diveSafetyFindings),
+        ]),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_log/data/repositories/dive_repository_impl.dart:198]: Correct, fixed in d8b828a, and chasing the cause turned up a second table you did not mention.

This is the third gap of the same kind on this tick, so the root cause is worth stating: a DiveSummary is assembled from THREE queries, not one. The paginated summary SELECT, a batched tag fetch joining tags and dive_tags, and a batched dive-type fetch over dive_dive_types. I built the tick from the first query's readsFrom alone, which made the two batched fetches structurally invisible to it. That is why this has arrived in instalments: trips and sites, then safety findings, now tags.

So dive_dive_types is in as well. The dive-type badges come from that junction, and a type assignment change had exactly the staleness you describe for tags. The type NAMES resolve through their own provider, so only the junction is needed for those.

Each of the three new tables has its own test and is independently mutation-checked: removing any one of diveTags, tags or diveDiveTypes fails its test and no other.

The doc comment now leads with why the set is larger than it looks, rather than just listing the tables, so the next person adding a field to the summary has a reason to check the tick instead of a list to copy.

File: test/core/database/query_plan_test.dart [hunk 13]
```diff
@@ -57,6 +59,15 @@ void main() {
     expect(p, contains('USING INDEX'));
   });
 
+  test('the trip-header join looks trips up, never scans them', () async {
+    // The list's group headers (#1193) read four scalars off trips. This is
+    // the hottest query in the app, so the join has to resolve by key: a
+    // SCAN here would cost one trips pass per page.
+    final p = await plan(db, _summariesPage1Sql);
+    expect(p, isNot(contains('SCAN trips')));
+    expect(p, contains('SEARCH t'));
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply test/core/database/query_plan_test.dart]: Correct, and worse than described. Fixed in 4ac8c28.

I printed the real plan rather than reasoning about the format, and it settles it:

    SEARCH d USING INDEX idx_dives_favorite (diver_id=?)
    SEARCH t USING INDEX sqlite_autoindex_trips_1 (id=?) LEFT-JOIN
    USE TEMP B-TREE FOR ORDER BY

SQLite names the alias and never the table, so `isNot(contains("SCAN trips"))` could not match anything the planner emits. It was not a weak assertion, it was a vacuous one: it would have passed a full scan. It now matches the alias with a word boundary, so an output that does name the table is still caught, and it is mutation-checked by making the join non-indexable, which the old assertion passed and the new one fails.

That same output exposed a second hole nobody flagged: there is no row for the sites join at all. The fixture selected only `t.name`, so SQLite dropped the `dive_sites` join as unused, and the guard was describing a query shape the app never runs. The fixture now mirrors the real SELECT and the sites join has its own assertion.

The two naming points from the same review are in as well: `divesChangeSub` is now `listChangeSub`, since it listens to `watchDiveListChanges` rather than the dives-only tick, and the test that said "still not paused by the sort" while asserting it is paused has been renamed.

**Comment by ericgriffin** [comment]
The suppressed finding about Intl.defaultLocale is fixed in cdc867b. No thread to reply on, so recording it here.

It was a gap in my own earlier locale fix rather than a fresh issue. I pinned `MaterialApp.locale`, which settles the l10n strings, and assumed that covered the header. It does not: the date range goes through intl `DateFormat`, which resolves against `Intl.defaultLocale`, a process global that `MaterialApp.locale` never touches. The `Jun` assertion was riding on intl's implicit `en_US` fallback, or on whatever another test left behind in the global.

Pinned in `setUp` and restored in `tearDown` so the global stays contained. No `initializeDateFormatting` call, unlike the pure unit tests in this repo: `GlobalMaterialLocalizations.delegate` initializes date formatting for widget tests.

The verification is the part worth writing down, because the obvious check does not work. Setting the global to `de` leaves the test passing, since German abbreviates June as "Juni" and `textContaining("Jun")` still matches. Under `fr` it fails with `Found 0 widgets with text containing Jun`, which is what actually proves the pin is load-bearing. A mutation that does not discriminate would have had me conclude the pin was decorative.

Only this one test asserts on a formatted date; the others assert on site names, counts and menu labels, which the MaterialApp locale pin already covers.
